app/scripts/lib/state-corruption-errors.ts

```diff
@@ -0,0 +1,97 @@
+import log from 'loglevel';
+import { escape as lodashEscape } from 'lodash';
+import {
+  SUPPORT_LINK,
+  VAULT_RECOVERY_LINK,
+} from '../../../shared/lib/ui-utils';
+import {
+  getErrorHtmlBase,
+  getLocaleContext,
+  setupLocale,
+} from '../../../shared/lib/error-utils';
+import {
+  MISSING_VAULT_ERROR,
+  CORRUPTION_BLOCK_CHECKSUM_MISMATCH,
+} from '../../../shared/constants/errors';
+import { switchDirectionForPreferredLocale } from '../../../shared/lib/switch-direction';
+import getFirstPreferredLangCode from './get-first-preferred-lang-code';
+
+export type ErrorLike = {
+  message: string;
+  name: string;
+  stack?: string;
+};
+
+export const METHOD_DISPLAY_STATE_CORRUPTION_ERROR =
+  'displayStateCorruptionError';
+
+export const KNOWN_STATE_CORRUPTION_ERRORS = new Set([
+  MISSING_VAULT_ERROR,
+  CORRUPTION_BLOCK_CHECKSUM_MISMATCH,
+]);
+
+export async function getStateCorruptionErrorHtml(
+  vaultRecoveryLink: string,
+  currentLocale?: string,
+  supportLink?: string,
+) {
+  let response, preferredLocale: string;
+  if (currentLocale) {
+    preferredLocale = currentLocale;
+    response = await setupLocale(currentLocale);
+  } else {
+    preferredLocale = await getFirstPreferredLangCode();
+    response = await setupLocale(preferredLocale);
+  }
```

Comment:
Nit: Can we return `getFirstPreferredLangCode()` in `currentLocale: _state?.data?.PreferencesController?.currentLocale,` when it is undefined? So we can simplify 
```suggestion
  response = await setupLocale(currentLocale);
```

Reply:
this function is mostly a copy+paste of `getErrorHtml` from `shared/lib/error-utils.js`, so I didn't think about trying to optimize the logic. good idea!

Reply:
So I realized that a lot of this is error logic actually copy + pasted from `shared/lib/error-utils.js`. I'd like to refactor it all after this PR lands, because it will end up changing a lot of code. I started on the work here: https://github.com/MetaMask/metamask-extension/pull/32160/files#diff-5a9d964db9d7c2fe207e21470884b2710e54723e0c1e6e6f7fd264706604fd6c